packager/media/chunking/chunking_handler.cc

```diff
@@ -151,6 +153,9 @@ Status ChunkingHandler::EndSegmentIfStarted() const {
   auto segment_info = std::make_shared<SegmentInfo>();
   segment_info->start_timestamp = segment_start_time_.value();
   segment_info->duration = max_segment_time_ - segment_start_time_.value();
+  segment_info->segment_index = current_segment_index_ +
+                                num_segments_before_last_cue_ +
+                                chunking_params_.start_segment_number - 1;
```

Comment:
There is a potential behavior change.

1. In the original code, the segment_index starts with 0 and is always monotonically increasing. 
2. In the new code, the segment_index is related to the segment timestamp.

Can we move the behavior change to a separate PR and focus this PR on start segment number?

So in this PR, we can just set the initial segment_index to chunking_params_.start_segment_number - 1 and increases by 1 for every new segment.

The comment applies to text_chunker below too.

Reply:
Sure. Let's do that.